### PR TITLE
feat(core): quick submit: stacked commits

### PR DESCRIPTION
**Pull request by rdunnington**
* Quick Submit now has no requirement to switch back to main when making multiple submits in a row. Users may commit as many times as desired before syncing main. This is achieved using git worktree to spin up a local copy of the repo, without inflating the LFS stub files. The worktree copy is rebased on top of latest main to ensure any changes in the local branch are smoothly merged with the changes in main, which avoids limitations in GitHub's ability to detect true conflicts when pushing commits that have already flowed through the merge queue.
* Removed PushOp as most of the logic was for pre-quick submit style pushes.

### CODE REVIEW

File: core/src/clients/git.rs [hunk 1]
```diff
@@ -463,6 +487,52 @@ impl Git {
...
+                    should_log_stdout: false,
+                    return_complete_error: true,
+                    lfs_mode: LfsMode::Stubs,
+                },
+            )
+            .await?;
+
+        let dir_regex = Regex::new(r"^worktree (.+)").unwrap();
+        let sha_regex = Regex::new(r"^HEAD (.+)").unwrap();
+        let branch_regex = Regex::new(r"^(branch|detached)\s*(.+)?").unwrap();
```
Comment (rudoi) [inline core/src/clients/git.rs:513]: Generally with static expressions we've tossed them in a `lazy_static!` though i should say that rust 1.70 has a solution to this that doesn't require a dependency: https://blog.rust-lang.org/2023/06/01/Rust-1.70.0.html#oncecell-and-oncelock

Non-blocking feedback either way.

File: core/src/clients/git.rs [hunk 2]
```diff
@@ -507,6 +577,10 @@ impl Git {
         // disable clone protection
         cmd.env("GIT_CLONE_PROTECTION_ACTIVE", "false");
 
+        if opts.lfs_mode == LfsMode::Stubs {
+            cmd.env("GIT_LFS_SKIP_SMUDGE", "1");
```
Comment (rudoi) [inline core/src/clients/git.rs:581]: Awesome. I can't help but think this new opt is going to help us with a lot of stuff in the future.

File: friendshipper/src/routes/source/submit/+page.svelte [hunk 3]
```diff
@@ -87,13 +86,7 @@
...
 	$: commitMessageValid = validateCommitMessage($commitMessage);
 
-	$: canSubmit =
-		$selectedFiles.length > 0 &&
-		get(commitMessage) !== '' &&
-		commitMessageValid &&
-		!loading &&
-		activeQuickSubmitPull === undefined &&
-		$repoConfig?.trunkBranch === $repoStatus?.branch;
+	$: canSubmit = $selectedFiles.length > 0 && get(commitMessage) !== '' && commitMessageValid;
```
Comment (rudoi) [inline friendshipper/src/routes/source/submit/+page.svelte:89]: This feels SO good haha.

**Review by rudoi — DISMISSED** [review]
All looks good to me. Super wonderful how quickly this came together :D

**Review by chris-believer — APPROVED** [review]
:fire:

File: friendshipper/src/routes/source/submit/+page.svelte (hunk 3 above)
Comment (rdunnington) [reply friendshipper/src/routes/source/submit/+page.svelte:89]: haha yes!

File: core/src/clients/git.rs (hunk 1 above)
Comment (rdunnington) [reply core/src/clients/git.rs:513]: Yeah I can do this

**Review by rudoi — APPROVED** [review]

**Pull request merged**